src/v/archival/ntp_archiver_service.cc

```diff
@@ -314,13 +314,21 @@ ss::future<cloud_storage::download_result> ntp_archiver::download_manifest() {
     gate_guard guard{_gate};
     retry_chain_node fib(
       _manifest_upload_timeout, _cloud_storage_initial_backoff, &_rtcnode);
-    retry_chain_logger ctxlog(archival_log, fib, _ntp.path());
-    vlog(ctxlog.debug, "Downloading manifest for {}", _ntp);
     auto path = _manifest.get_manifest_path();
     auto key = cloud_storage::remote_manifest_path(
       std::filesystem::path(std::move(path)));
+    auto bucket = _bucket;
+    if (_partition && _partition->is_read_replica_mode_enabled()) {
+        bucket = _partition->get_read_replica_bucket();
+        vlog(
+          _rtclog.debug,
+          "Downloading manifest for the read replica, bucket: {}",
```

Comment:
So here we are substituting the bucket name for the manifest download. Do we need to do the same for segment download or is it already working correctly?